snippets/SharedDestroy.hpp

```diff
@@ -2,6 +2,12 @@
 template <typename HandleType>
 class SharedHandleTraits;
 
+// Silence the function cast warnings.
+#ifdef __GNUC__
```

Comment:
Is silencing this warning the correct thing to do?
Can't the problem be resolved without ignoring that warning?

Reply:
Unfortunately not, as gcc compiler doesn't recognize the function cast properly. No other compilers complain, and the cast there is perfectly fine, as we only need to upcast the Dispatcher pointer.

Reply:
Which casts does it complain about?

Reply:
m_destroy in SharedDestroy in any class there, something like:
`reinterpret_cast<decltype( m_destroy )>( static_cast<DestroyFunctionPointerType<Dispatcher>>( &DestructorType::free ) )`
it complains about that explicit reinterpret_cast there.
The thing is, only difference there is the dispatcher pointer, but we already store the correct one, and only thing we care about is type erasure. So reinterpret_cast does just that type erasure.

Reply:
Agreed, that should be ok.